tests/php/test-amp-crowdsignal-embed-handler.php

```diff
@@ -82,6 +82,10 @@ public function test_conversion( $url, $expected, $oembed_response ) {
 		add_filter(
 			'pre_http_request',
 			static function ( $pre, $r, $request_url ) use ( $oembed_response ) {
+				if ( in_array( 'external-http', $_SERVER['argv'], true ) ) {
```

Comment:
This would check if the `external-http` testsuite is being run, and to act accordingly. Cleaner solutions for this are welcomed.

Reply:
The ideal seems to be to obtain the name of the test suite bring run via some phpunit API. I'm not seeing this, however. Perhaps @schlessera is familiar with one.

Reply:
Due to how PHPUnit is designed there is no way of dynamically adding or removing tests at runtime, As far as I am concerned.

Reply:
Maybe you could use the `@require` annotation to do this.
Not tested, but it might work to use something like `@require function do_external_http_request` and then conditionally define the `do_external_http_request()` function during bootstrap or in a listener...

See https://phpunit.readthedocs.io/en/8.4/incomplete-and-skipped-tests.html#incomplete-and-skipped-tests-requires-tables-api

Reply:
Using a listener wouldn't be possible as the signature of  the interface to implement changes between each major PHPUnit release, so I'll opt for the defining the function when bootstrapping.

Reply:
Using the `@requires` annotation wouldn't allow for the flexibility of mocking a HTTP request during the test. The aim is not having to duplicate the logic just to not mock the request and maintain both instances.